src/sentry/snuba/metrics/extraction.py

```diff
@@ -1,81 +1,116 @@
 RuleCondition = Union["LogicalRuleCondition", "ComparingRuleCondition", "NotRuleCondition"]
 
-_SNUBA_TO_RELAY_FIELDS = {
-    "contexts[geo.country_code]": "event.geo.country_code",
-    "http_method": "event.http.method",
+
+# Maps from Discover's field names to event protocol paths.
+# See Relay's ``FieldValueProvider`` for supported fields.
+_SEARCH_TO_PROTOCOL_FIELDS = {
+    # Top-level fields
     "release": "event.release",
-    "transaction_op": "event.transaction.op",
-    "transaction_status": "event.transaction.status",
-    "duration": "event.duration",
-    "measurements[cls]": "event.measurements.cls",
-    "measurements[fcp]": "event.measurements.fcp",
-    "measurements[fid]": "event.measurements.fid",
-    "measurements[fp]": "event.measurements.fp",
-    "measurements[lcp]": "event.measurements.lcp",
-    "measurements[ttfb]": "event.measurements.ttfb",
-    "measurements[ttfb.requesttime]": "event.measurements.ttfb.requesttime",
-    # TODO(ogi): Support fields whose resolution returns a function
-    # "browser.name": "event.browser.name",
-    # "http.status_code": "event.http.status_code",
-    # "os.name": "event.os.name",
+    "dist": "event.dist",
+    "environment": "event.environment",
+    "transaction": "event.transaction",
+    "platform": "event.platform",
+    # User
+    "user.email": "event.user.email",
+    "user.id": "event.user.id",
+    "user.ip_address": "event.user.ip_address",
+    "user.name": "event.user.name",
+    "user.segment": "event.user.segment",
+    # Subset of context fields
+    "device.name": "event.contexts.device.name",
+    "device.family": "event.contexts.device.family",
+    "os.name": "event.contexts.os.name",
+    "os.version": "event.contexts.os.version",
+    "transaction.op": "event.contexts.trace.op",
+    # Computed fields
+    "transaction.duration": "event.duration",
+    "release.build": "event.release.build",
+    "release.package": "event.release.package",
+    "release.version": "event.release.version.short",
+    # Tags, measurements, and breakdowns are mapped by the converter
+    # TODO: Required but yet unsupported by Relay
+    # "geo.country_code": None,
+    # "transaction.status": None,
+    # "http.method": None,
+}
+
+# Maps from Discover's syntax to Relay rule condition operators.
+_SEARCH_TO_RELAY_OPERATORS: Dict[str, "CompareOp"] = {
+    "=": "eq",
+    "!=": "eq",  # combined with external negation
+    "<": "lt",
+    "<=": "lte",
+    ">": "gt",
+    ">=": "gte",
 }
 
-_SNUBA_TO_METRIC_AGGREGATES: Dict[str, Optional[MetricOperationType]] = {
+# Maps plain Discover functions to metric aggregation functions. Derived metrics
+# are not part of this mapping.
+_SEARCH_TO_METRIC_AGGREGATES: Dict[str, Optional[MetricOperationType]] = {
     "count": "sum",
     "avg": "avg",
-    "quantile(0.50)": "p50",
-    "quantile(0.75)": "p75",
-    "quantile(0.95)": "p95",
-    "quantile(0.99)": "p99",
     "max": "max",
-    # not supported yet
```

Comment:
The unsupported functions are now listed in a TODO inside `_extract_field_info`, since we need to write more procedural code to support derived metrics.